### PR TITLE
fixing some build warnings

### PR DESCRIPTION
**Pull request by brettsam**
Cleaning up some low-hanging fruit to get our warnings down a little. Also fixing the documentation file issue.

### CODE REVIEW

File: src/DotNetWorker/DotNetWorker.csproj [hunk 1]
```diff
@@ -8,7 +8,7 @@
     <AssemblyName>Microsoft.Azure.Functions.Worker</AssemblyName>
     <RootNamespace>Microsoft.Azure.Functions.Worker</RootNamespace>
     <SignAssembly>true</SignAssembly>
-    <DocumentationFile>$(OutputPath)\$(AssemblyName).xml</DocumentationFile>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
```
Comment (brettsam) [inline src/DotNetWorker/DotNetWorker.csproj:11]: Following the advice here -- https://docs.microsoft.com/en-us/dotnet/csharp/codedoc, which I *think* should work for us.

Turns out `$(OutputPath)` isn't always set here -- so you end up with `\....xml`, which resolves to c:\

Comment (fabiocav) [reply src/DotNetWorker/DotNetWorker.csproj:11]: Have you checked the result? It does get packaged, right?

File: src/DotNetWorker/Http/HttpRequestDataExtensions.cs [hunk 2]
```diff
@@ -19,7 +19,7 @@ public static class HttpRequestDataExtensions
         /// </summary>
         /// <param name="request">The request from which to read.</param>
         /// <param name="encoding">The encoding to use when reading the string. Defaults to UTF-8</param>
-        /// <returns>A <see cref="Task{string?}"/> that represents the asynchronous read operation.</returns>
+        /// <returns>A task that represents the asynchronous read operation.</returns>
```
Comment (fabiocav) [inline src/DotNetWorker/Http/HttpRequestDataExtensions.cs:22]: We should be able to have a generic type reference here, correct?

File: src/DotNetWorker/IFunctionActivator.cs [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 namespace Microsoft.Azure.Functions.Worker
 {
-    public interface IFunctionActivator
+    internal interface IFunctionActivator
```
Comment (fabiocav) [inline src/DotNetWorker/IFunctionActivator.cs:8]: Do we expect this to change? Should it stay public?

File: src/DotNetWorker/Http/HttpRequestDataExtensions.cs (hunk 2 above)
Comment (brettsam) [reply src/DotNetWorker/Http/HttpRequestDataExtensions.cs:22]: I think you can do Task<T> -- but not Task<specifictype> -- I checked some other official docs and they just say "A task..." -- but let me look for other examples.

File: src/DotNetWorker/IFunctionActivator.cs (hunk 3 above)
Comment (brettsam) [reply src/DotNetWorker/IFunctionActivator.cs:8]: I'd rather keep it internal by default and open it up in a follow-up.

File: src/DotNetWorker/DotNetWorker.csproj (hunk 1 above)
Comment (brettsam) [reply src/DotNetWorker/DotNetWorker.csproj:11]: Yes, just verified.

Comment (fabiocav) [reply src/DotNetWorker/DotNetWorker.csproj:11]: 👍🏼

File: src/DotNetWorker/Http/HttpRequestDataExtensions.cs (hunk 2 above)
Comment (fabiocav) [reply src/DotNetWorker/Http/HttpRequestDataExtensions.cs:22]: Was this maybe just an issue with using the alias? (`string` vs `String`)?

File: src/DotNetWorker/IFunctionActivator.cs (hunk 3 above)
Comment (fabiocav) [reply src/DotNetWorker/IFunctionActivator.cs:8]: Not opposed to that, but this is one of the key interfaces that I can see people needing to use

**Review by fabiocav — APPROVED** [review]
Approving, but if you can check on the Task comment, that would be good

File: src/DotNetWorker/Http/HttpRequestDataExtensions.cs (hunk 2 above)
Comment (brettsam) [reply src/DotNetWorker/Http/HttpRequestDataExtensions.cs]: oh... it was the `?` throwing it off. fixed it up.

**Pull request merged**